### ISSUE TITLE
Feature request: allow a single label for a group of lines

### ISSUE THREAD

**Issue by kinyatoride** [proplot-dev/proplot#423]
### Description

Hi, is it possible to allow a single label for a group of lines? Currently, we need to pass the same number of labels for plotted elements. 
Here, I am showing an example of the wished behavior.

```python
import numpy as np
import proplot as pplt

state = np.random.RandomState(51423)
x = np.linspace(0, 5, 20)
y1 = x[:, np.newaxis] + state.rand(20, 10)
y2 = x[::-1, np.newaxis] + state.rand(20, 10)

fig, ax = pplt.subplots()
ax.plot(y1, c='r', label='y1')
ax.plot(y2, c='b', label='y2')
ax.legend()
```

**Expected behavior**
![output](https://user-images.githubusercontent.com/61028484/235486911-2e074ef1-260b-482a-97f2-01ad84181543.png)

**Comment by lukelbd** [comment]
Makes sense. Maybe that should be the default behavior when `label` is used instead of `labels` (currently they are aliases)? For now here's a clumsy workaround:

```python
import numpy as np
import proplot as pplt

state = np.random.RandomState(51423)
x = np.linspace(0, 5, 20)
y1 = x[:, np.newaxis] + state.rand(20, 10)
y2 = x[::-1, np.newaxis] + state.rand(20, 10)

fig, ax = pplt.subplots()
ax.plot(y1, c='r', label=['y1'] + 9 * [None])
ax.plot(y2, c='b', label=['y2'] + 9 * [None])
ax.legend()
```

![iTerm2 UKZCDv tmp58qbkxgz](https://user-images.githubusercontent.com/19657652/235490979-e3d933be-e743-4582-a82c-3b97c29e2eb3.png)

**Comment by kinyatoride** [comment]
On second thought, this might cause unexpected behavior. We need to specify a color (and other properties) for the group of lines. If we pass a single label without specifying a color, it won't know which color to use for the legend. 
We could use the property of the first line, but I don't know if this is a good idea.

**Comment by lukelbd** [comment]
Hmm yeah I can see that. And actually this is matplotlib's default behavior :/... not sure if any better than current behavior of raising an error. Think I'll close then

```python
import numpy as np
import matplotlib.pyplot as plt

state = np.random.RandomState(51423)
x = np.linspace(0, 5, 20)
y1 = x[:, np.newaxis] + state.rand(20, 10)
y2 = x[::-1, np.newaxis] + state.rand(20, 10)

fig, ax = plt.subplots()
ax.plot(y1, c='r', label='y1')
ax.plot(y2, c='b', label='y2')
ax.legend()
```

![iTerm2 EYQgA4 tmpjutgkpek](https://user-images.githubusercontent.com/19657652/235495201-441c78cb-0ae7-4b90-a52a-8bd3766bb7d6.png)

**Comment by lukelbd** [comment]
On second thought... even though this matplotlib example is weird it probably makes more sense to assign identical labels to each line rather than raising an error. That way users can easily assign labels, but prevent repeated legend entries by (e.g.) manually passing the line handles to `legend()`. I'll re-open
